### ISSUE TITLE
customize typesPrefix for enums

### ISSUE THREAD

**Issue by veeramarni** [dotansimha/graphql-code-generator#1980]
**Is your feature request related to a problem? Please describe.**

Is there a way we can customize the `prefix` to be used only for `types` and leave `enums` as it is?

**Describe the solution you'd like**

Right now when we set `typesPrefix: I`, it prefixes all the `types` as well as `enums`. But it doesn't make sense for us to have `I` prefix to enums. So we don't want `I` prefixes to `enums`.

Like `namingConvention` which can be customized for `typeNames ` and `enumValues `, can we have the same feature for `typesPrefix` for `typeNames` and `enumNames`?

**Comment by dotansimha** [comment]
As we mentioned before, the `types` in `typesPrefix` is for the TypeScript `type` and not for the GraphQL `type`. 
There is a similar issue to this one: https://github.com/dotansimha/graphql-code-generator/issues/1712

I guess adding `enumsPrefix` as additional configuration will support both use-cases, because this way it's possible to set `typesPrefix: I` and `enumsPrefix: ""` when you want to reset it.

**Comment by veeramarni** [comment]
similarly, if we could also have `interfacePrefix` option so if we already have `I` prefix in the graphql field we don't want to add any additional prefix.

**Comment by veeramarni** [comment]
@1.4.0 works for me.

**Comment by cml391** [comment]
I'm wondering what the status of this is. I'm trying to do that same thing using 1.4.0 and it doesn't appear to be working for me if I use `interfacePrefix` or `enumsPrefix` in conjunction with `typesPrefix`. Is this something that might be released soon?

**Comment by dotansimha** [comment]
@cml391 `enumsPrefix` was not implemented, it was just a suggestion. It was closed because it's tracked here: https://github.com/dotansimha/graphql-code-generator/issues/1712

**Comment by dotansimha** [comment]
@cml391 you can now (with the latest alpha, we'll release it soon) use `enumPrefix: false` to remove the prefix from enums.

**Comment by cml391** [comment]
@dotansimha thanks for following up! For context, I have been using the `declarationKind: 'interface'` and `typesPrefix: I`, but I want to not prefix enums with 'I' (for obvious reasons). I tried to use `enumPrefix: false` with my current schema using the latest alpha of the typescript package. I noticed that when I had used enums in my Query arguments, the RootQueryType[...]Args types still put a prefix on the enum so I ended up with a bunch of missing "can not find name" type errors since the enum definition and the type name used in args didn't match. I'm not sure if this is something you're aware of, but wanted to flag it to you in case you had missed that

**Comment by ardatan** [comment]
@cml391 Do we have a chance to have a reproduction to get on the same page?

**Comment by cml391** [comment]
@ardatan ya, so my configurations is:
```
plugins:
      - typescript
    config:
      skipTypename: true
      declarationKind: 'interface'
      typesPrefix: I
      enumPrefix: false
      constEnums: true
      scalars:
        DateTime: string
        Time: string
        Date: string
```

and an example of a schema piece that isn't working is

```
enum SuggestionType {
  concern
  goal
}

type Suggestion {
  id: ID!
  userId: ID!
  suggestionType: SuggestionType!
  text: String!
}

type RootQueryType {
    suggestionsForUser(
        userId: ID!
        suggestionType: SuggestionType!
    ): [Suggestion!]
}
```

This turns into: 
```
export interface ISuggestion {
  id: Scalars['ID'],
  userId: Scalars['ID'],
  suggestionType: SuggestionType,
  text: Scalars['String']
}

export const enum SuggestionType {
  Concern = 'concern',
  Goal = 'goal'
};

export interface IRootQueryType {
  suggestionsForUser?: Maybe<Array<ISuggestion>>,
}

export interface IRootQueryTypeSuggestionsForUserArgs {
  userId: Scalars['ID'],
  suggestionType: ISuggestionType
}
```
So in the enum definition, the `I` prefix is removed, but when it is used `IRootQueryTypeSuggestionsForUserArgs`, the prefix is not removed and there are errors saying `cannot find name 'ISuggestionType'` since the names don't match.

**Comment by dotansimha** [comment]
@cml391 this was fixed in 1.6.0 :)
